incubator/hnc/pkg/reconcilers/test_helpers_test.go

```diff
@@ -168,6 +173,26 @@ func deleteObject(ctx context.Context, kind string, nsName, name string) {
 	ExpectWithOffset(1, k8sClient.Delete(ctx, inst)).Should(Succeed())
 }
 
+// cleanupObjects makes a best attempt to cleanup all objects created from makeObject.
+func cleanupObjects(ctx context.Context) {
+	for i := 0; i < 3 && len(createdObjects) > 0; i++ {
+		if i > 0 {
+			time.Sleep(100 * time.Millisecond)
+		}
+
+		retryList := []*unstructured.Unstructured{}
+		for _, obj := range createdObjects {
+			err := k8sClient.Delete(ctx, obj)
```

Comment:
Instead of doing retry by ourselves, does it make sense to use [Eventually](https://onsi.github.io/gomega/#eventually) and let the function takes care of the retry.

Reply:
Let me investigate that. I think it's even better if we just say Eventually we want the error code to be NotFound.. something like: `Eventually(errors.IsNotFound(k8sClient.Delete(ctx, obj))).Should(BeTrue())`

The downside to that is that is that the first call will have no error and we'll have to wait for the poll to come back in the happy case.

Thoughts?

Reply:
I took your suggestion and went with `Eventually` in the event the initial delete doesn't work